kivy/core/window/_window_sdl2.pyx

```diff
@@ -394,6 +394,14 @@ cdef class _WindowSDL2Storage:
     def set_window_pos(self, x, y):
         SDL_SetWindowPosition(self.win, x, y)
 
+    def set_window_opacity(self, opacity):
+        SDL_SetWindowOpacity(self.win, opacity)
```

Comment:
From SDL2 docs https://wiki.libsdl.org/SDL2/SDL_SetWindowOpacity function `SDL_SetWindowOpacity` returns `-1` if opacity is not supported or (another) negative value in case of an error. We should use `SDL_GetError` to get the error message and log it.